src/cmd/run.rs

```diff
@@ -203,30 +203,60 @@ async fn poll_integration<D: Database>(
     db: &D,
     integration: &Integration,
     sender: &Queue<ScanGitVCSReference>,
+) -> Result<(), Error> {
+    let poll_interval = integration.poll_interval().as_duration();
+    loop {
+        if let Err(err) = execute_poll_integration(db, integration, sender).await {
+            warn!("Unable to poll '{integration}': {err:#?}");
+        }
+
+        // Now wait for the next poll time.
+        // The fact that we poll, _then_ wait for the poll time, means that the actual
+        // time at which the poll occurs will slowly creep forward by whatever time
+        // it takes to perform the poll.
+        //
+        // This is considered okay, because we're interpreting "poll interval" as
+        // "poll at most this often"; i.e. it's used primarily as a rate limiting feature
+        // than as a "track changes this often" feature.
+        //
+        // The alternative would be starting the clock at the top of the loop,
+        // which while doable could (in the worst case) allow for endlessly
+        // polling if the poll takes at least as long as the interval.
+        //
+        // If we decide to make polling more consistent, [`tokio::time::interval`]
+        // is most likely the correct way to implement it.
+        info!("Next poll interval for '{integration}' in {poll_interval:?}");
+        tokio::time::sleep(poll_interval).await;
+    }
+}
+
+#[tracing::instrument(skip_all)]
+async fn execute_poll_integration<D: Database>(
+    db: &D,
+    integration: &Integration,
+    sender: &Queue<ScanGitVCSReference>,
 ) -> Result<(), Error> {
     // We use this in a few places and may send it across threads, so just clone it locally.
     let remote = integration.remote().to_owned();
-    let poll_interval = integration.poll_interval().as_duration();
 
     // [`Retry`] needs a function that runs without any arguments to perform the retry, so turn the method into a closure.
     let get_references = || async {
         match integration.references().await {
             Ok(success) => Ok(success),
             Err(err) => {
-                warn!("attempt to poll integration at {remote} failed: {err:#}");
+                warn!("Unable to poll integration at {remote}: {err:#}");
```

Comment:
Based on your previous comments regarding how we should write our logging messages,  I went ahead and changed this existing warning message as well